components/worktemplates/components/preview/section.tsx

```diff
@@ -3,51 +3,138 @@
 
 import React from 'react';
 import {useIntl} from 'react-intl';
-import styled from 'styled-components';
+import classnames from 'classnames';
 
-interface PreviewSectionProps {
-    className?: string;
-    message: string;
-    items: Array<{id: string; name: string; illustration?: string}>;
+import {useSelector} from 'react-redux';
 
-    onUpdateIllustration: (illustration: string) => void;
-}
+import {isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
 
-const PreviewSection = styled((props: PreviewSectionProps) => {
-    const {formatMessage} = useIntl();
+import './section.scss';
 
-    const updateIllustration = (e: React.MouseEvent<HTMLAnchorElement>, illustration: string) => {
-        e.preventDefault();
+interface GenericPreviewSectionProps {
+    items: Array<{id: string; name?: string; illustration?: string}>;
+    onUpdateIllustration?: (illustration: string) => void;
+    className?: string;
+    message?: string;
+    id?: string;
+}
 
-        props.onUpdateIllustration(illustration);
-    };
+type IntegrationPreviewSectionItemsProps = {id: string; name?: string; category?: string; description?: string; icon?: string; installed?: boolean}
+interface IntegrationPreviewSectionProps {
+    items: IntegrationPreviewSectionItemsProps[];
+    className?: string;
+    message?: string;
+    id?: string;
+}
+
+const viewIntegrations = (props: GenericPreviewSectionProps | IntegrationPreviewSectionProps) => {
+    if (props.id === 'integrations') {
+        const integrationsProps = props as IntegrationPreviewSectionProps;
+        return (
+            <IntegrationsPreview
+                items={integrationsProps.items}
+            />);
+    }
+    const genericProps = props as GenericPreviewSectionProps;
+    return (
+        <GenericPreview
+            items={props.items}
+            onUpdateIllustration={genericProps.onUpdateIllustration}
+        />);
+};
 
+const PreviewSection = (props: GenericPreviewSectionProps | IntegrationPreviewSectionProps) => {
+    const {formatMessage} = useIntl();
     return (
         <div className={props.className}>
             <p>
                 {props.message}
             </p>
             <span className='included-title'>{formatMessage({id: 'worktemplates.preview.section.included', defaultMessage: 'Included'})}</span>
-            <ul>
-                {props.items.map((c) => (
-                    <li key={c.id}>
-                        <a
-                            href='#'
-                            onClick={(e) => updateIllustration(e, c.illustration || '')}
+            { viewIntegrations(props) }
+
+        </div>
+    );
+};
+
+const IntegrationsPreview = ({items}: IntegrationPreviewSectionProps) => {
+    const isAdmin = useSelector(isCurrentUserSystemAdmin);
```

Comment:
We should look specifically for a role with access to plugin install and not just sysadmins.